### DISCUSSION TITLE
Streams support

### DISCUSSION THREAD

**Discussion by GrzybDev** [flet-dev/flet#4307] [Ideas]
Hello,

The thing that would absolutely benefit Flet is streams support.

Currently, `Audio`, `AudioRecorder`, `FilePicker` controls are rather limited in functionality, in their current state, the controls are assuming that all developers want to for example upload the file to specific pre-signed URL, rather than parse file directly in for example client's browser. Similar story with Audio/AudioRecorder where we are expected to either record to file or play from specific path/url.

Adding for example `content` field of type `BytesIO` to `FilePickerFile` would make FilePicker on the `web` actually useful, because in current form it's practically useless - assuming you actually don't want to upload anything to server, or the server expects custom POST/PUT request (not the pre-signed URL).

**Comment by FeodorFitsner** [comment]
Is it possible to process a selected file on a browser side without uploading it to a server?

**Reply by GrzybDev** [reply]
In current version of flet, I'm afraid there's not. But that would absolutely benefit flet :)

**Reply by GrzybDev** [reply]
FilePicker is the most obvious example, the other controls that would benefit from more... control (haha) are for example Audio/AudioRecorder ones. In current state, `AudioRecorder` allows me to only record microphone audio to file. With streams, it should be possible to record to python's bytearray (or at least make a callback function when in buffer there's minimum amount of bytes required)

**Reply by GrzybDev** [reply]
@FeodorFitsner After small modification in both file-picker.dart and file-picker.py I can expose the "bytes" of the selected file in python - so it's definitely possible, but the problem with my method is that it immediately loads selected file to memory (in order to pass it to python) - that's dangerous and not really recommended (what if user selects 100GB file?), that's why Flet really needs to implement some kind of streams support.

Currently in FilePicker dart code we're actually using streams internally (for the upload part), but we don't expose the streams at all to python-side.

**Comment by FeodorFitsner** [comment]
This might be useful re: files: https://github.com/pyodide/pyodide/issues/679#issuecomment-637504864
